### ISSUE TITLE
[Issue]: Lightbox modal not displayed

### ISSUE THREAD

**Issue by Fenrir200678** [vladmandic/automatic#368]
### Feature description

Hi there!

Would you mind adding an image zoom / lightbox modal for generated images like in A1111? I really love this fork, but not beeing able to zoom into my generated images without saving and open them locally unfortunately is a deal breaker for me. 

Or maybe it is already there and simply not working for me? 🤔

### Version Platform Description

fresh install of latest version, Windows 22H2, RTX 4090, latest Edge

**Comment by vladmandic** [comment]
if that functionality is present in core A1111 repo, I did not remove it - and yes, i'd like to see it working as well.
please provide exact workflow and what happens when using a) A1111, b) this repo.

**Comment by Fenrir200678** [comment]
Sure. In A1111 I just click on the generated image...
![1](https://user-images.githubusercontent.com/15671097/233796150-1f1119be-0164-443d-9048-72deebfb0ff3.png)

... and it opens this lightbox modal to zoom the image.
![2](https://user-images.githubusercontent.com/15671097/233796168-1bd3b94b-16b4-426a-9a5c-2bbda35f10d2.png)

Basically its just this lightboxModal div which gets activated. I didn't install this as an extra extension as far as I know, so I guess it has to be in the A1111 core.
![3](https://user-images.githubusercontent.com/15671097/233796199-0e3c3506-2781-4dd5-ad1d-2b36ea11b35a.png)

In your repo nothing happens when I click the image. There is also no lightbox modal to see in the HTML.
![4](https://user-images.githubusercontent.com/15671097/233796244-4e208acb-869c-40e6-843b-75264d078e1a.png)

**Comment by vladmandic** [comment]
i wasn't sure what you were referring to as this is a very basic feature that works for 99% of users.
i just checked on chrome/edge/firefox and no issues.
there is no lightbox model in html since it gets created ondemand - and it didn't in your case for unknown reason.
what does your browser console log say?

**Comment by Fenrir200678** [comment]
Nothing really special, except some translation warnings. No error is thrown on load, on generation or on click of the image. 
![image](https://user-images.githubusercontent.com/15671097/233796639-c7b3c723-2d7c-4b74-9170-29ce240497bf.png)

**Comment by Fenrir200678** [comment]
I also got no special extensions installed except the aspec ratio helper in case your wondering.
![image](https://user-images.githubusercontent.com/15671097/233796771-85ecf536-6ee7-463c-9654-2b405e50ad29.png)

**Comment by AlUlkesh** [comment]
I made a PR on the A1111 side for this issue a while ago: https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/9060

**Comment by vladmandic** [comment]
nice to know, i'll merge it here, just need to fix some merge conflicts with it:
<https://github.com/vladmandic/automatic/pull/385>

**Comment by vladmandic** [comment]
i've made a mess of that merge, but finally cleaned it up and merged.

**Comment by Fenrir200678** [comment]
Unfortunately still doesn't work for me after git pulling latest version. At least the div#lightboxModal gets rendered out now, but the event listener doesn't seem to work. So nothing happens on click, the modal doesn't get populated and shown. :/

![Screenshot 2023-04-23 171025](https://user-images.githubusercontent.com/15671097/233847888-c54a7be2-2ac5-41b7-9851-1dedb76d29d6.png)

**Comment by vladmandic** [comment]
yup, i see your inspector shows it as created, but display is set to none, so its not visible.
what happens if you manually change css `display` property from `none` to `block`

**Comment by Fenrir200678** [comment]
That's what I tried first. :) Then just the opaque modal background gets displayed, but since the actual image isn't populated, no zoomed image is there.

![image](https://user-images.githubusercontent.com/15671097/233849719-c10ca280-e4f7-4221-818b-fce83258159c.png)

**Comment by vladmandic** [comment]
i'm at a loss, this will require adding some debug code to javascript so we can capture what is going on. need to place on backlog for now, i hope someone else can pick it up quicker?

**Comment by Fenrir200678** [comment]
Figured it out by myself after digging in the code.
![image](https://user-images.githubusercontent.com/15671097/233851778-7853d6f9-81dd-41cc-9a24-fe38523bb5a1.png)

In imageviewer.js the eventlistener is setup. If opts.js_modal_lightbox isn't true it just returns without displaying the lightbox. After further digging around I found out, that there is an option "Enable full page image viewer" in the User Interface tab which wasn't checked. I'm quite sure, that I didn't uncheck this (not 100% sure though), so it maybe is unchecked by default?

![image](https://user-images.githubusercontent.com/15671097/233851966-de173e8d-ddca-4eef-b9f2-76f255800c4b.png)

Anyhow, problem solved now. Thank you anyways for the support! :)
